connector/src/main/java/org/geysermc/connector/utils/AdvancementsUtils.java

```diff
@@ -0,0 +1,147 @@
+import org.geysermc.common.window.response.SimpleFormResponse;
+import org.geysermc.connector.network.session.GeyserSession;
+import org.geysermc.connector.network.translators.chat.MessageTranslator;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AdvancementsUtils {
+
+    // Used in UpstreamPacketHandler.java
+    public static final int ADVANCEMENTS_MENU_FORM_ID = 1341;
+    public static final int ADVANCEMENTS_LIST_FORM_ID = 1342;
+
+
+    /**
+     * Build a form for the given session with all advancement categories
+     *
+     * @param session The session to build the form for
+     */
+    public static SimpleFormWindow buildMenuForm(GeyserSession session) throws NullPointerException {
+        // Cache the language for cleaner access
+        String language = session.getClientData().getLanguageCode();
+        // Created menu window for advancement categories
+        SimpleFormWindow window = new SimpleFormWindow(LocaleUtils.getLocaleString("gui.advancements", language), "");
+        int baseId = 0;
+
+        for (Map.Entry<String, Advancement> advancement : session.getStoredAdvancements().entrySet()) {
+            String title = advancement.getValue().getDisplayData().getTitle().toString();
+            String description = advancement.getValue().getDisplayData().getDescription().toString();
+            String[] idAndTitle = {advancement.getValue().getId(), advancement.getValue().getDisplayData().getTitle().toString()};
+
+            if (advancement.getValue().getId().endsWith("root")) {
+                window.getButtons().add(new FormButton(MessageTranslator.convertMessage(title, language) + " - " + MessageTranslator.convertMessage(description, language)));
+                session.getButtonIdsToIdAndTitleAdvancements().put(baseId++, idAndTitle);
+
+            }
+
+
+        }
+
+        return window;
+    }
+
+
+    /**
+     * Handle the menu form response
+     *
+     * @param session  The session that sent the response
+     * @param response The response string to parse
+     * @return True if the form was parsed correctly, false if not
+     */
+    public static boolean handleMenuForm(GeyserSession session, String response) {
+        // Cache the language for cleaner access
+        String language = session.getClientData().getLanguageCode();
+        SimpleFormWindow menuForm = (SimpleFormWindow) session.getWindowCache().getWindows().get(ADVANCEMENTS_MENU_FORM_ID);
+        menuForm.setResponse(response);
+        SimpleFormResponse formResponse = (SimpleFormResponse) menuForm.getResponse();
+        int clickedButton = -1;
+
+        if (formResponse != null && formResponse.getClickedButton() != null) {
+            clickedButton = formResponse.getClickedButtonId();
+
+        }
+        StringBuilder content = new StringBuilder();
+        String[] idAndTitle = session.getButtonIdsToIdAndTitleAdvancements().get(clickedButton);
```

Comment:
`clickedButton` doesn't need to be stored in a variable since its only set and accessed once.
The empty line should be after the `}` not before it.